### PR TITLE
getdate: improve year conversion

### PR DESCRIPTION
**Pull request by ikerexxe**
libmisc/getdate.y: improve year conversion

### CODE REVIEW

File: libmisc/getdate.y [hunk 1]
```diff
@@ -395,7 +401,8 @@ relunit	: tUNUMBER tYEAR_UNIT {
 
 number	: tUNUMBER
           {
-	    if ((yyHaveTime != 0) && (yyHaveDate != 0) && (yyHaveRel == 0))
+	    if ((yyHaveTime != 0 || $1 >= 100) && !yyHaveYear
+		&& (yyHaveDate != 0) && (yyHaveRel == 0))
```
Comment (hallyn) [inline libmisc/getdate.y:405]: I think this looks correct, but it's not exactly easy to parse.
What do you think about moving this baroque check into its own
helper function?

File: libmisc/getdate.y [hunk 2]
```diff
@@ -802,7 +809,7 @@ yylex (void)
 	  return LookupWord (buff);
 	}
       if (c != '(')
-	return *yyInput++;
+	return (unsigned char)*yyInput++;
```
Comment (hallyn) [inline libmisc/getdate.y:812]: The function returns int.  Why this cast to unsigned char?

**Review by hallyn — COMMENTED** [review]
Thanks!  If you feel my comments are off please let me know and I'll merge.

**Comment by ikerexxe** [comment]
I'll take those concerns into account and propose a new patch. Thanks for the feedback.

File: libmisc/getdate.y (hunk 1 above)
Comment (ikerexxe) [reply libmisc/getdate.y]: I'd like to do that but my understanding of yacc is low and I fear it'll take me a lot of time to do it. Do you mind if we keep the change as it is?

File: libmisc/getdate.y (hunk 2 above)
Comment (ikerexxe) [reply libmisc/getdate.y]: This didn't make any sense. Sorry for including it. I've restored the previous version.

**Comment by hallyn** [comment]
The commit says "improve year conversion".  What change precisely is it intended to make?

(Would be reassuring if there were some testcases for getdate. :)

**Comment by hallyn** [comment]
Ping - can you elaborate (in the commit message) on how this improves the conversion?

**Comment by ikerexxe** [comment]
Sorry, I haven't been able to work on this topic lately and I don't think I'll be able to do it before the end of the year. I'll close the PR and open a new one when I have time to clarify it.

**Pull request closed without merging**